Start/stop UX: solana-localnet/start-*.sh, solana-localnet/stop-*.sh, deletions of old scripts

eydelrivero opened the pull request:
# Start/stop UX: `solana-localnet/start-*.sh`, `solana-localnet/stop-*.sh`, deletions of old scripts

## 📝 Summary

Improves the UX with docker and podman by adding scritps for easy management of the lifecycle of the stack.

## 🎯 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔧 Configuration change
- [ ] 🚀 Performance improvement
- [ ] ♻️ Code refactoring (no functional changes)

## 🔍 Scope & Complexity

**Please keep PRs focused and small for faster reviews.**

- [ ] This PR changes **fewer than 400 lines** of code (excluding generated files)
- [x] This PR addresses **only one logical change** or feature
- [x] This PR can be **reviewed in under 30 minutes**
- [x] This PR does **not mix multiple types of changes** (e.g., refactoring + new features)

If any of the above are unchecked, consider breaking this PR into smaller, focused changes.

## 📋 Changes Made

Detailed list of changes:

- Added script for starting and stopping localnet with docker
- Added script for starting and stopping localnet with podman
- Added script for starting and stopping the prod ansible-control with docker

## 🧪 Testing

- [ ] Pre-commit hooks pass (`pre-commit run --all-files`)
- [x] Manual testing completed
- [x] Existing functionality verified unchanged
- [ ] Documentation updated (if applicable)

### Test Details

# First clean your docker and podman containers and volumens. Then use this workflow:

```sh
cd solana-localnet

# start docker localnet
./start-localnet-docker.sh
# check docker localnet is up
docker ps -a

# clean docker localnet
./stop-localnet-docker.sh
# check docker localnet is removed
docker ps -a

# start podman localnet
./start-localnet-podman.sh
# check podman localnet is up
podman ps -a 

# clean podman localnet
./stop-localnet-podman.sh
# check podman localnet is removed
podman ps -a

# start docker prod ansible-control
./start-prod-ansible-control.
# check docker prod ansible-control is up
docker ps -a

# clean docker prod ansible-control
./stop-prod-ansible-control.sh
# check docker prod ansible-control is removed
docker ps -a
```

## 📚 Documentation

- [ ] Updated relevant documentation
- [x] Added/updated comments for complex logic
- [ ] README updated (if applicable)
- [ ] No documentation changes needed

## 📝 Review Notes

---

## For Reviewers

**Estimated review time:** ⏱️ 30X minutes]

**Focus areas:**
- [x] Logic correctness
- [ ] Security implications
- [ ] Performance impact
- [ ] Documentation completeness

MexicanSparker commented on solana-localnet/start-host-monitoring.sh:
```diff
@@ -0,0 +1,41 @@
+done
+
+echo "$ENGINE compose version:"
+$COMPOSE_BIN --version
```
Launching `docker compose --version` did not work, it shows the command help. The command help indicates `docker compose version`
Launching `podman compose --version` works but the command help indicates `podman compose version`
Essentially removing the `--`

eydelrivero replied:
Already fixed.

MexicanSparker commented:
Tests carried out (continuation):

1. clean docker localnet
<img width="5120" height="2880" alt="02 - clean docker localnet - also stop host-monitoring worked" src="https://github.com/user-attachments/assets/aec8221f-42b1-4699-a55b-9dbd29ad97a2" />
2. start podman localnet
  a. When the podman build starts again `host-bravo` displays with a red cross with an error:
  <img width="2830" height="2308" alt="03a - start podman localnet - Start build" src="https://github.com/user-attachments/assets/101cf311-8f5b-4c60-a692-fa6b6c212d20" />
  b. In the end everything finishes well and the conatiner is attached without issues:
  <img width="2830" height="2308" alt="03b - start podman localnet - Container attached successfully" src="https://github.com/user-attachments/assets/f3713be0-f60b-4980-8ee2-72a7b68f095b" />
  c. I have completed a full validator setup without any issue
3. clean podman localnet
  a. The command executed well but while all containers stopped, `host-bravo` and `host-charlie` were not removed.
  <img width="5120" height="2880" alt="04 - clean podman localnet - full stop but partial removal" src="https://github.com/user-attachments/assets/81a98b46-1126-4492-8d72-4aaad1b4e847" />
4. start docker prod ansible-control
  a. No issues. It worked like a charm. Still pending if we have all the tooling but I can still setup an empty metal box and see if the playbooks installs all the needed tooling that we don't need to add to the image, pending (next testnet v2 is good opportunity)
  b. Tested connectivity to a live server to verify, all good.
  <img width="3064" height="2488" alt="05 - start docker prod ansible-control - worked" src="https://github.com/user-attachments/assets/42127ccf-ffaf-4507-aa6e-231d3f5285ba" />
4. clean docker prod ansible-control. WORKED!

eydelrivero commented:
> Tests carried out:
> 
> 1. start docker localnet
>    a. When the docker build starts I receive an error about `host-bravo`, not sure why.

This error is been there since forever and it can be safely ignored. I've added a fix for it anyways.

> b. After finishing the build a recap summarizes the built containers and `host-bravo` looks like it is ok. Right aferwards I receive a docker command line help because the command should be `docker compose version` so by removing the `--` would solve it. 

I removed any call to `compose --version`.

> c. The container attach works very well, no issues.

Good.

> d. As routine checks I have run a full validator setup and worked well, also ran the tests and the rust v2 tests ran well but two new isues appeared in the `iam_manager role tests` which I will open a different issue to fix that. e. After starting a fresh localnet with 1a) I did not attached the container, instead ran script `./start-host-monitoring.sh` and this was the outcome: 

OK.

MexicanSparker reviewed (approved):
👍
